cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -94,9 +135,11 @@ Result<DeclarationInfo> FromProto(const substrait::Rel& rel, const ExtensionSet&
         const substrait::ReadRel::NamedTable& named_table = read.named_table();
         std::vector<std::string> table_names(named_table.names().begin(),
                                              named_table.names().end());
-        ARROW_ASSIGN_OR_RAISE(compute::Declaration source_decl,
+        ARROW_ASSIGN_OR_RAISE(compute::Declaration no_emit_declaration,
```

Comment:
Can you revert this change?  From the perspective of `FromProto` I think this is still the "source_decl".  The name "no_emit_declaration" only really makes sense inside `ProcessEmit`.  If you need to differentiate I think you could have `source_no_emit` or `source_before_emit` but I'm not sure that is really needed.  The old name was fine I think.

Reply:
Sure, we can stick to that.